app/api/reviewit/merge_requests.rb

```diff
@@ -35,19 +35,26 @@ def same_patch?
 
       desc 'Create a new merge request'
       post do
-        MergeRequest.transaction do
-          mr = MergeRequest.new
-          diff = Diff.new(params[:diff])
-          mr.project = project
-          mr.author = current_user
-          mr.subject = diff.subject
-          mr.target_branch = params[:target_branch]
-          mr.add_patch(diff: diff,
-                       linter_ok: params[:linter_ok].true?,
-                       ci_enabled: params[:ci].true?)
-          mr.save!
-          MergeRequestMailer.created(mr).deliver_now
-          { mr_id: mr.id }
+
+        branch_exist = Git.new.branch_exist(project.repository, params[:target_branch])
+
+        if branch_exist
```

Comment:
`raise 'The target branch does not exist on repository.' unless branch_exist` would be better looking.